src/java/test/org/apache/zookeeper/test/KeyAuthClientTest.java

```diff
@@ -0,0 +1,131 @@
+        zk.create(path, null, Ids.CREATOR_ALL_ACL, CreateMode.PERSISTENT);
+        List<ACL> acls = zk.getACL(path, null);
+        LOG.debug("Node: "+path+" Test:"+testName+" ACLs:");
+        for (ACL acl : acls) {
+          LOG.debug("  "+acl.toString());
+        }
+      } catch (Exception e) {
+          LOG.debug("  EXCEPTION THROWN", e);
+      }
+    }
+
+    public void testPreAuth() throws Exception {
+        ZooKeeper zk = createClient();
+        zk.addAuthInfo("key", "25".getBytes());
+        try {
+            createNodePrintAcl(zk, "/pre", "testPreAuth");
+            zk.setACL("/", Ids.CREATOR_ALL_ACL, -1);
+            zk.getChildren("/", false);
+            zk.create("/abc", null, Ids.CREATOR_ALL_ACL, CreateMode.PERSISTENT);
+            zk.setData("/abc", "testData1".getBytes(), -1);
+            zk.create("/key", null, Ids.CREATOR_ALL_ACL, CreateMode.PERSISTENT);
+            zk.setData("/key", "5".getBytes(), -1);
+            Thread.sleep(1000);
+        } catch (KeeperException e) {
+            Assert.fail("test failed :" + e);
+        }
+        finally {
+            zk.close();
+        }
+    }
+
+    public void testMissingAuth() throws Exception {
+        ZooKeeper zk = createClient();
+        try {
+            zk.getData("/abc", false, null);
+            Assert.fail("Should not be able to get data");
+        } catch (KeeperException correct) {
+            // correct
+        }
+        try {
+            zk.setData("/abc", "testData2".getBytes(), -1);
+            Assert.fail("Should not be able to set data");
+        } catch (KeeperException correct) {
+            // correct
+        } finally {
+            zk.close();
+        }
+    }
+
+    public void testValidAuth() throws Exception {
+        ZooKeeper zk = createClient();
+        // any multiple of 5 will do...
+        zk.addAuthInfo("key", "25".getBytes());
+        try {
+            createNodePrintAcl(zk, "/valid", "testValidAuth");
+            zk.getData("/abc", false, null);
+            zk.setData("/abc", "testData3".getBytes(), -1);
+        } catch (KeeperException.AuthFailedException e) {
+            Assert.fail("test failed :" + e);
+        } finally {
+            zk.close();
+        }
+    }
+
+    public void testValidAuth2() throws Exception {
+        ZooKeeper zk = createClient();
+        // any multiple of 5 will do...
+        zk.addAuthInfo("key", "125".getBytes());
+        try {
+            createNodePrintAcl(zk, "/valid2", "testValidAuth2");
+            zk.getData("/abc", false, null);
+            zk.setData("/abc", "testData3".getBytes(), -1);
+        } catch (KeeperException.AuthFailedException e) {
+            Assert.fail("test failed :" + e);
+        } finally {
+            zk.close();
+        }
+    }
+
+    @Test
+    public void testAuth() throws Exception {
+        // NOTE: the tests need to run in-order, and older versions of
```

Comment:
I think it is fine if you have a sequence of steps in your test case, but I really see this as a single test case with multiple steps. I suggest we remove the `test` prefix from the methods to make clear that they aren't independent test cases.